### ISSUE TITLE
create ~/.config/oil/ if missing at invocation

### ISSUE THREAD

**Issue by wbob** [oilshell/oil#345]
I wondered that history is empty after exiting the shell and entering again. If I `mkdir -p ~/.config/oil` myself, `history_osh` is then created and written to. Creating the directory is not part of the install instructions (oil-0.6.pre22). I think when invoking the shell interactively it could check if the directory exists and is writeable, and if not issues a warning or goes ahead and creates the directory.

**Comment by andychu** [comment]
Yes good point!  At the very least, I'll definitely document this somewhere.  It might make sense to create it too, not sure yet.

**Comment by andychu** [comment]
@wbob For now, I documented this as the very first thing in the manual:

https://github.com/oilshell/oil/blob/master/doc/osh-manual.md

And I also linked the manual from the /release/$VERSION page (or it will be in the next release)

It still might be a good idea to create ~/.config/oil automatically and avoid this.  But I don't think other shells do that -- they leave it up to the distro.  If it becomes a problem, we can revisit it.

Thanks for the report!
